libs/sdk-ui-dashboard/src/model/commandHandlers/widgets/exportInsightWidgetHandler.ts

```diff
@@ -0,0 +1,95 @@
+// (C) 2021 GoodData Corporation
+import { SagaIterator } from "redux-saga";
+import { call, put, select } from "redux-saga/effects";
+import { serializeObjRef } from "@gooddata/sdk-model";
+import { IExecutionResult } from "@gooddata/sdk-backend-spi";
+import invariant from "ts-invariant";
+
+import { ExportInsightWidget } from "../../commands";
+import { invalidArgumentsProvided } from "../../events/general";
+import {
+    DashboardInsightWidgetExportResolved,
+    insightWidgetExportRequested,
+    insightWidgetExportResolved,
+} from "../../events/insight";
+import {
+    selectExecutionResultByRef,
+    selectIsExecutionResultReadyForExportByRef,
+    selectIsExecutionResultExportableToCsvByRef,
+    selectIsExecutionResultExportableToXlsxByRef,
+} from "../../state/executionResults/executionResultsSelectors";
+import { DashboardContext } from "../../types/commonTypes";
+import { createExportFunction, IExtendedExportConfig } from "@gooddata/sdk-ui";
+import { PromiseFnReturnType } from "../../types/sagas";
+
+async function performExport(
+    executionResult: IExecutionResult,
+    config: IExtendedExportConfig,
+): Promise<string> {
+    const exporter = createExportFunction(executionResult);
+    const result = await exporter(config);
+    return result.uri;
+}
+
+function* validateSettingsAndPermissions(
+    ctx: DashboardContext,
+    cmd: ExportInsightWidget,
+): SagaIterator<void> {
+    const { config, ref } = cmd.payload;
+
+    let canExport = false;
+    if (config.format === "csv") {
+        canExport = yield select(selectIsExecutionResultExportableToCsvByRef(ref));
+    }
+
+    if (config.format === "xlsx") {
+        canExport = yield select(selectIsExecutionResultExportableToXlsxByRef(ref));
+    }
+
+    if (!canExport) {
+        throw invalidArgumentsProvided(
+            ctx,
+            cmd,
+            `The widget with ref: ${serializeObjRef(
+                ref,
+            )} cannot be exported because the feature is disabled or the user does not have the necessary permissions.`,
+        );
+    }
+}
+
+export function* exportInsightWidgetHandler(
+    ctx: DashboardContext,
+    cmd: ExportInsightWidget,
+): SagaIterator<DashboardInsightWidgetExportResolved> {
+    const { config, ref } = cmd.payload;
+
+    yield put(insightWidgetExportRequested(ctx, ref, config, cmd.correlationId));
+
+    const isExportable: ReturnType<ReturnType<typeof selectIsExecutionResultReadyForExportByRef>> =
```

Comment:
Could you separate isExportable into method and name it like validateExecResult ... 

Than just call validateExecResult same way as validateSettingsAndPermissions